renku/core/models/jsonld.py

```diff
@@ -632,10 +632,11 @@ def asjsonld(self):
         source.update(asjsonld(self))
         return source
 
-    def to_yaml(self):
+    def to_yaml(self, path=None):
         """Store an instance to the referenced YAML file."""
         jsonld_ = self.asjsonld()
-        write_yaml(path=self.__reference__, data=jsonld_)
+        path = path or self.__reference__
```

Comment:
So far the intended use of this was to wrapp calls to `to_yaml` into `with with_reference():`. I'm not sure if that has any benefit over supplying the path directly (It does some check to make sure the reference wasn't changed). I think we should update to this new version everywhere that is currently wrapped in `with_reference` to make the code more readable.

Reply:
I've created an issue for this: https://github.com/SwissDataScienceCenter/renku-python/issues/1361.